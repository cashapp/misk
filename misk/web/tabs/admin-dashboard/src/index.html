<!DOCTYPE html>
<html>

<head>
  <meta charset="UTF-8" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=0" />
  <meta http-equiv="X-UA-Compatible" content="ie=edge" />
  <link rel="shortcut icon" type="image/x-icon" href="/static/favicon.ico" />
</head>

<body>
  <div id="admin-dashboard"></div>

  <!-- Misk Libraries -->
  <link rel="stylesheet" type="text/css" href="/@misk/common/styles.css" />
  <script type="text/javascript" src="/@misk/common/vendors.js" preload></script>
  <script type="text/javascript" src="/@misk/common/common.js" preload></script>
  <script type="text/javascript" src="/@misk/core/core.js" preload></script>
  <script type="text/javascript" src="/@misk/simpleredux/simpleredux.js" preload></script>

  <!-- Tab Admin Dashboard -->
  <script type="text/javascript" src="/_admin/tab_admin-dashboard.js"></script>
</body>

</html>
