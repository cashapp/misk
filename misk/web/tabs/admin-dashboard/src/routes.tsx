import * as React from "react"
import { Route, Switch } from "react-router"
import { AdminDashboardContainer } from "./containers"

const routes = (
  <span>
    <Switch>
      <Route component={AdminDashboardContainer} />
    </Switch>
  </span>
)

export default routes
