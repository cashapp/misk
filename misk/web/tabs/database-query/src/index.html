<!DOCTYPE html>
<html>

<head>
  <meta charset="UTF-8" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=0" />
  <meta http-equiv="X-UA-Compatible" content="ie=edge" />
</head>

<body>
  <div id="<%= htmlWebpackPlugin.options.slug %>"></div>

  <!-- Misk Libraries -->
  <link rel="stylesheet" type="text/css" href="/@misk/common/styles.css" />
  <script type="text/javascript" src="/@misk/common/vendors.js" preload></script>
  <script type="text/javascript" src="/@misk/common/common.js" preload></script>
  <script type="text/javascript" src="/@misk/core/core.js" preload></script>
  <script type="text/javascript" src="/@misk/simpleredux/simpleredux.js" preload></script>

  <!-- Tabs -->
</body>

</html>
