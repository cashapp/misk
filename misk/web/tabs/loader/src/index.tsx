import { IMiskAdminTabs } from "@misk/common"
import { connectRouter, routerMiddleware } from "connected-react-router"
import { createBrowserHistory } from "history"
import * as React from "react"
import * as ReactDOM from "react-dom"
import { AppContainer } from "react-hot-loader"
import { Provider } from "react-redux"
import { applyMiddleware, compose, createStore } from "redux"
import createSagaMiddleware from "redux-saga"
import App from "./App"
import rootReducer from "./reducers"
import rootSaga from "./sagas"

const history = createBrowserHistory()
const sagaMiddleware = createSagaMiddleware()

const composeEnhancer: typeof compose = (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose
const store = createStore(
  connectRouter(history)(rootReducer),
  composeEnhancer(
    applyMiddleware(
      sagaMiddleware,
      routerMiddleware(history),
    ),
  ),
)

/**
 * Starts the rootSaga which forks off instances of all sagas used to receive and process actions as they are dispatched (./sagas/index.ts)
 */
sagaMiddleware.run(rootSaga)

const render = () => {
  ReactDOM.render(
    <AppContainer>
      <Provider store={store}>
        <App history={history} />
      </Provider>
    </AppContainer>,
    document.getElementById("loader")
  )
}

render()

// Hot reloading
if (module.hot) {
  // Reload components
  module.hot.accept("./App", () => {
    render()
  })

  // Reload reducers
  module.hot.accept("./reducers", () => {
    store.replaceReducer(connectRouter(history)(rootReducer))
  })
}