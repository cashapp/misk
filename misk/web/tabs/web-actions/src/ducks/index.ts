import {
  dispatchSimpleRedux,
  IDispatchSimpleRedux,
  ISimpleReduxImmutableState,
  ISimpleReduxState,
  SimpleReduxReducer,
  SimpleReduxSaga,
  simpleRootSelector,
  watchSimpleReduxSagas
} from "@misk/simpleredux"
import {
  connectRouter,
  LocationChangeAction,
  RouterState
} from "connected-react-router"
import { History } from "history"
import { AnyAction, combineReducers, Reducer } from "redux"
import { all, fork } from "redux-saga/effects"
import {
  dispatchWebActions,
  IDispatchWebActions,
  IWebActionsImmutableState,
  IWebActionsState,
  watchWebActionsSagas,
  WebActionsReducer
} from "./webActions"
export * from "./webActions"

/**
 * Redux Store State
 */
export interface IState {
  router: Reducer<RouterState, LocationChangeAction>
  simpleRedux: ISimpleReduxState
  webActions: IWebActionsState
  webActionsRaw: IWebActionsImmutableState
}

/**
 * Dispatcher
 */
export interface IDispatchProps
  extends IDispatchSimpleRedux,
    IDispatchWebActions {}

export const rootDispatcher: IDispatchProps = {
  ...dispatchSimpleRedux,
  ...dispatchWebActions
}

/**
 * State Selectors
 */
export const rootSelectors = (state: IState) => ({
  router: state.router,
  simpleRedux: simpleRootSelector<IState, ISimpleReduxImmutableState>(
    "simpleRedux",
    state
  ),
  webActions: simpleRootSelector<IState, IWebActionsImmutableState>(
    "webActions",
    state
  ),
  webActionsRaw: state.webActionsRaw
})

/**
 * Reducers
 */
export const rootReducer = (history: History): Reducer<any, AnyAction> =>
  combineReducers({
    router: connectRouter(history),
    simpleRedux: SimpleReduxReducer,
    webActions: WebActionsReducer,
    webActionsRaw: WebActionsReducer
  })

/**
 * Sagas
 */
export function* rootSaga(): SimpleReduxSaga {
  yield all([fork(watchWebActionsSagas), fork(watchSimpleReduxSagas)])
}

/**
 * Map Dispatch/State to Props
 */
export const mapStateToProps = (state: IState) => rootSelectors(state)

export const mapDispatchToProps: IDispatchProps = {
  ...rootDispatcher
}
