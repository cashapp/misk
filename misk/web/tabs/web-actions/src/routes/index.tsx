import * as React from "react"
import { Route, Switch } from "react-router"
import { TabContainer as OldTabContainer } from "../containers"
import TabContainer from "../rewrite/TabContainer"

const routes = (
  <div>
    <Switch>
      <Route path="/_admin/web-actions/" component={OldTabContainer} />
      <Route path="/_admin/web-actions-beta/" component={TabContainer} />
      {/* Do not include a Route without a path or it will display during on all tabs */}
    </Switch>
  </div>
)

export default routes
