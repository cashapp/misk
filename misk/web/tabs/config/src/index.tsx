///<reference types="webpack-env" />
import { IWindow } from "@misk/common"
import { connectRouter, routerMiddleware } from "connected-react-router"
import { createBrowserHistory } from "history"
import * as React from "react"
import * as ReactDOM from "react-dom"
import { AppContainer } from "react-hot-loader"
import { Provider } from "react-redux"
import { applyMiddleware, compose, createStore } from "redux"
import createSagaMiddleware from "redux-saga"
import App from "./App"
import { rootReducer, rootSaga } from "./ducks"

export * from "./components"
export * from "./containers"

const tabSlug = "config"

const Window = window as IWindow

Window.Misk.History = Window.Misk.History || createBrowserHistory()
const history = Window.Misk.History
const sagaMiddleware = createSagaMiddleware()

const composeEnhancer: typeof compose = Window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose
const store = createStore(
  connectRouter(history)(rootReducer),
  composeEnhancer(
    applyMiddleware(
      sagaMiddleware,
      routerMiddleware(history),
    ),
  ),
)

/**
 * Starts the rootSaga which forks off instances of all sagas used to receive and process actions as they are dispatched (./sagas/index.ts)
 */
sagaMiddleware.run(rootSaga)

const render = () => {
  ReactDOM.render(
    <AppContainer>
      <Provider store={store}>
        <App history={history} />
      </Provider>
    </AppContainer>,
    document.getElementById(tabSlug)
  )
}

render()

// Hot reloading
if (module.hot) {
  // Reload components
  module.hot.accept("./App", () => {
    render()
  })

  // Reload reducers
  module.hot.accept("./ducks", () => {
    store.replaceReducer(connectRouter(history)(rootReducer))
  })
}