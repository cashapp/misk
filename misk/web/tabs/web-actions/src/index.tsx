import { createApp, createIndex } from "@misk/core"
import * as Ducks from "./ducks"
import routes from "./routes"
export * from "./containers"

createIndex("web-actions", createApp(routes), Ducks)
