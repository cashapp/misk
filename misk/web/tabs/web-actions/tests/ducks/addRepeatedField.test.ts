import HTTPMethod from "http-method-enum"
import { addRepeatedField, generateTypesMetadata } from "../../src/ducks"
import { nonTypedActionAPI, testTypes } from "../testUtilities"

describe("Add a repeated field", () => {
  it("addRepeated", () => {
    let typesMetadata = generateTypesMetadata({
      ...nonTypedActionAPI,
      httpMethod: HTTPMethod.POST,
      requestType: "repeatedShort",
      types: testTypes
    })
    const parentId = typesMetadata.get("0").idChildren.first() as string
    expect(typesMetadata.get(parentId).idChildren.size).toBe(1)
    typesMetadata = addRepeatedField(testTypes, typesMetadata, parentId)
    expect(typesMetadata.get(parentId).idChildren.size).toBe(2)
    typesMetadata = addRepeatedField(testTypes, typesMetadata, parentId)
    expect(typesMetadata.get(parentId).idChildren.size).toBe(3)
    expect(typesMetadata).toMatchSnapshot()
  })
  it("addNestedRepeated", () => {
    let typesMetadata = generateTypesMetadata({
      ...nonTypedActionAPI,
      httpMethod: HTTPMethod.POST,
      requestType: "repeatedNestedRepeatedShort",
      types: testTypes
    })
    const parentId = typesMetadata.get("0").idChildren.first() as string
    expect(typesMetadata.size).toBe(6)
    expect(typesMetadata.get(parentId).idChildren.size).toBe(1)
    typesMetadata = addRepeatedField(testTypes, typesMetadata, parentId)
    expect(typesMetadata.get(parentId).idChildren.size).toBe(2)
    const nestedRepeatedId2 = typesMetadata
      .get(typesMetadata.get(parentId).idChildren.last())
      .idChildren.first() as string
    expect(typesMetadata.size).toBe(10)
    expect(typesMetadata.get(nestedRepeatedId2).idChildren.size).toBe(1)
    typesMetadata = addRepeatedField(
      testTypes,
      typesMetadata,
      nestedRepeatedId2
    )
    expect(typesMetadata.get(nestedRepeatedId2).idChildren.size).toBe(2)
    expect(typesMetadata.size).toBe(13)
    expect(typesMetadata).toMatchSnapshot()
  })
})
