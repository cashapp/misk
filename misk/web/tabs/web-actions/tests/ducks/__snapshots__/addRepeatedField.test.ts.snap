// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add a repeated field addNestedRepeated 1`] = `
Immutable.OrderedMap {
  "0": Object {
    "dirtyInput": false,
    "id": "0",
    "idChildren": Immutable.OrderedSet [
      "5",
    ],
    "idParent": "0",
    "name": "",
    "repeated": false,
    "serverType": null,
    "typescriptType": null,
  },
  "5": Object {
    "dirtyInput": false,
    "id": "5",
    "idChildren": Immutable.OrderedSet [
      "6",
      "10",
    ],
    "idParent": "0",
    "name": "Nested Repeated Short Field",
    "repeated": true,
    "serverType": "nestedRepeatedShort",
    "typescriptType": null,
  },
  "6": Object {
    "dirtyInput": false,
    "id": "6",
    "idChildren": Immutable.OrderedSet [
      "7",
    ],
    "idParent": "5",
    "name": "Nested Repeated Short Field",
    "repeated": false,
    "serverType": "nestedRepeatedShort",
    "typescriptType": null,
  },
  "7": Object {
    "dirtyInput": false,
    "id": "7",
    "idChildren": Immutable.OrderedSet [
      "8",
    ],
    "idParent": "6",
    "name": "Nested Short Field",
    "repeated": false,
    "serverType": "repeatedShort",
    "typescriptType": null,
  },
  "8": Object {
    "dirtyInput": false,
    "id": "8",
    "idChildren": Immutable.OrderedSet [
      "9",
    ],
    "idParent": "7",
    "name": "Repeated Short Field",
    "repeated": true,
    "serverType": "Short",
    "typescriptType": null,
  },
  "9": Object {
    "dirtyInput": false,
    "id": "9",
    "idChildren": Immutable.OrderedSet [],
    "idParent": "8",
    "name": "Repeated Short Field",
    "repeated": false,
    "serverType": "Short",
    "typescriptType": "number",
  },
  "10": Object {
    "dirtyInput": false,
    "id": "10",
    "idChildren": Immutable.OrderedSet [
      "11",
    ],
    "idParent": "5",
    "name": "Nested Repeated Short Field",
    "repeated": false,
    "serverType": "nestedRepeatedShort",
    "typescriptType": null,
  },
  "11": Object {
    "dirtyInput": false,
    "id": "11",
    "idChildren": Immutable.OrderedSet [
      "12",
      "14",
    ],
    "idParent": "10",
    "name": "Nested Short Field",
    "repeated": false,
    "serverType": "repeatedShort",
    "typescriptType": null,
  },
  "12": Object {
    "dirtyInput": false,
    "id": "12",
    "idChildren": Immutable.OrderedSet [
      "13",
    ],
    "idParent": "11",
    "name": "Repeated Short Field",
    "repeated": true,
    "serverType": "Short",
    "typescriptType": null,
  },
  "13": Object {
    "dirtyInput": false,
    "id": "13",
    "idChildren": Immutable.OrderedSet [],
    "idParent": "12",
    "name": "Repeated Short Field",
    "repeated": false,
    "serverType": "Short",
    "typescriptType": "number",
  },
  "14": Object {
    "dirtyInput": false,
    "id": "14",
    "idChildren": Immutable.OrderedSet [
      "15",
    ],
    "idParent": "11",
    "name": "Nested Short Field",
    "repeated": false,
    "serverType": "repeatedShort",
    "typescriptType": null,
  },
  "15": Object {
    "dirtyInput": false,
    "id": "15",
    "idChildren": Immutable.OrderedSet [
      "16",
    ],
    "idParent": "14",
    "name": "Repeated Short Field",
    "repeated": true,
    "serverType": "Short",
    "typescriptType": null,
  },
  "16": Object {
    "dirtyInput": false,
    "id": "16",
    "idChildren": Immutable.OrderedSet [],
    "idParent": "15",
    "name": "Repeated Short Field",
    "repeated": false,
    "serverType": "Short",
    "typescriptType": "number",
  },
}
`;

exports[`Add a repeated field addRepeated 1`] = `
Immutable.OrderedMap {
  "0": Object {
    "dirtyInput": false,
    "id": "0",
    "idChildren": Immutable.OrderedSet [
      "1",
    ],
    "idParent": "0",
    "name": "",
    "repeated": false,
    "serverType": null,
    "typescriptType": null,
  },
  "1": Object {
    "dirtyInput": false,
    "id": "1",
    "idChildren": Immutable.OrderedSet [
      "2",
      "3",
      "4",
    ],
    "idParent": "0",
    "name": "Repeated Short Field",
    "repeated": true,
    "serverType": "Short",
    "typescriptType": null,
  },
  "2": Object {
    "dirtyInput": false,
    "id": "2",
    "idChildren": Immutable.OrderedSet [],
    "idParent": "1",
    "name": "Repeated Short Field",
    "repeated": false,
    "serverType": "Short",
    "typescriptType": "number",
  },
  "3": Object {
    "dirtyInput": false,
    "id": "3",
    "idChildren": Immutable.OrderedSet [],
    "idParent": "1",
    "name": "Repeated Short Field",
    "repeated": false,
    "serverType": "Short",
    "typescriptType": "number",
  },
  "4": Object {
    "dirtyInput": false,
    "id": "4",
    "idChildren": Immutable.OrderedSet [],
    "idParent": "1",
    "name": "Repeated Short Field",
    "repeated": false,
    "serverType": "Short",
    "typescriptType": "number",
  },
}
`;
